### ISSUE TITLE
[Bug]: The SSH tunnel fails because of a messy `~/.ssh/config`

### ISSUE THREAD

**Issue by peterschmidt85** [dstackai/dstack#933]
Steps to reproduce:

1. Have a messy `~/.ssh/config` (containing some problematic settings)
2. Run `dstack server`
3. Define `.dstack.yml`
4. Run `dstack run .`

The server fails to establish the SSH tunnel:

```
DEBUG 2024-02-23T08:59:20.314 dstack._internal.core.services.ssh.tunnel SSH tunnel failed: b'ssh: connect to host california-a.tensordockmarketplace.com port 20499: Connection refused\r\n'
```

A possible reason:

After a long debugging session with @spott, we figured out that the problem goes away if the server adds `-F <home directory>/.dstack/ssh/config` to all the `ssh` commands aiming to establish a connection. This solution appears to resolve the issue because, in that case, `dstack` doesn't utilize the user's default SSH config, where some parts may cause the issue.

**Comment by spott** [comment]
[here](https://github.com/dstackai/dstack/pull/934) is a pull request to implement some of these changes.

**Comment by peterschmidt85** [comment]
Gonna be fixed in 0.16.1
